mesonbuild/backend/ninjabackend.py

```diff
@@ -2155,6 +2155,17 @@ def generate_rust_target(self, target: build.BuildTarget) -> None:
             self.generate_shsym(target)
         self.create_target_source_introspection(target, rustc, args, [main_rust_file], [])
 
+        if target.doctests:
+            from ..compilers.rust import RustCompiler
+            assert isinstance(rustc, RustCompiler)
+            assert target.doctests.target is not None
+            rustdoc = rustc.get_rustdoc(self.environment)
+            args = rustdoc.get_exe_args()
+            args += self.get_rust_compiler_args(target.doctests.target, rustdoc, target.rust_crate_type)
+            _, _, deps_args = self.get_rust_compiler_deps_and_args(target.doctests.target, rustdoc)
+            args += deps_args
+            target.doctests.cmd_args[0:0] = args.to_native() + [main_rust_file]
```

Comment:
It took me a minute to understand what as going on here. Is there are reason that you can't simply replace the `cmd_args` as `cmd_args = ...`? If we really need to mutate in place I think:
```python
target.doctests.cmd_args.clear()
target.doctests.cmd_args.extend(args.not_native())
target.doctests.cmd_args.append(main_rust_file)
```
is much clearer

Reply:
The `x[0:0] = ...` syntax is for *prepending* to a list (kinda like `extendleft` if this was a deque. I'll replace it with:

            target.doctests.cmd_args = args.to_native() + [main_rust_file] + target.doctests.cmd_args